sshd-common/src/main/java/org/apache/sshd/common/config/keys/KeyUtils.java

```diff
@@ -1196,4 +1204,13 @@ public static boolean compareSkEd25519Keys(SkED25519PublicKey k1, SkED25519Publi
                     && SecurityUtils.compareEDDSAPPublicKeys(k1.getDelegatePublicKey(), k2.getDelegatePublicKey());
         }
     }
+
+    public static String getHashAlgorithmType(String keyType) {
+        String hashAlgorithmType = CERT_TYPE_HASH_ALGORITHM_TYPE.get(keyType);
+        if (hashAlgorithmType != null) {
+            return hashAlgorithmType;
+        } else {
+            return keyType;
+        }
+    }
```

Comment:
Code can be simplified as `return CERT_TYPE_HASH_ALGORITHM_TYPE.getOrDefault(keyYpe, keyType)`